lib/abci/handlers/query/verifyChainlockQueryHandlerFactory.js

```diff
@@ -0,0 +1,40 @@
+const { ChainLock } = require('@dashevo/dashcore-lib');
+
+const {
+  abci: {
+    ResponseQuery,
+  },
+} = require('abci/types');
+
+const InvalidArgumentAbciError = require('../../errors/InvalidArgumentAbciError');
+
+/**
+ *
+ * @param {SimplifiedMasternodeList} simplifiedMasternodeList
+ * @return {verifyChainlockQueryHandler}
+ */
+function verifyChainlockQueryHandlerFactory(simplifiedMasternodeList) {
+  /**
+   * @typedef identityQueryHandler
+   * @param {Object} params
+   * @param {Object} data
+   * @param {Buffer} data.chainLock
+   * @return {Promise<ResponseQuery>}
+   */
+  async function verifyChainlockQueryHandler(params, { chainLock }) {
+    const chainlock = new ChainLock(chainLock);
+
+    if (!chainlock.verify(simplifiedMasternodeList.getStore())) {
+      throw new InvalidArgumentAbciError(
```

Comment:
I would pass chain lock as data param. Just pass `chainlock.toJSON()` as a second argument.

Reply:
Not entirely sure what you mean. You mean like this?
```
  async function verifyChainlockQueryHandler(params, data) {
    const chainlock = data.chainLock.toJSON();
```

Reply:
Nope. In `data`, the chainLock will be in binary format.

So you need to create a function:

`lib/core/decodeChainLock`
```js
function decodeChainLock(buffer) {
   return new ChainLock(buffer);
}
```

Then register it in `createDIContainer`:
```js
/**
 * Register Core
 */
container.register({
  ...
  decodeChainLock: asValue(decodeChainLock),
  ...
})
```

Since this function is registered in DI container you can use it as a dependency in `verifyChainlockQueryHandlerFactory`

```js
verifyChainlockQueryHandlerFactory(simplifiedMasternodeList, decodeChainLock) {
  ...
  async function verifyChainlockQueryHandler(params, data) {
    const chainlock = decodeChainLock(data.chainLock);
```

Doing that in your test you can pass a stub instead of `decodeChainLock` function and return a valid or not valid chain lock from this stub. You can find how to deal with stubs in other tests.

Reply:
Thank you, will do!